lib/rules/selector-not-notation/index.js

```diff
@@ -0,0 +1,195 @@
+'use strict';
+
+const isStandardSyntaxRule = require('../../utils/isStandardSyntaxRule');
+const isStandardSyntaxSelector = require('../../utils/isStandardSyntaxSelector');
+const parseSelector = require('../../utils/parseSelector');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const {
+	isPseudoClass: buggyIsPseudoClass, // cf https://github.com/postcss/postcss-selector-parser/blob/master/src/selectors/guards.js#L52
+	isAttribute,
+	isClassName,
+	isUniversal,
+	isIdentifier,
+	isTag,
+} = require('postcss-selector-parser');
+
+const ruleName = 'selector-not-notation';
+const messages = ruleMessages(ruleName, {
+	expected: (type) => `Expected ${type} :not() pseudo-class notation`,
+});
+const meta = { url: 'https://stylelint.io/user-guide/rules/list/selector-not-notation' };
+
+/** @typedef {import('postcss-selector-parser').Node} Node */
+/** @typedef {import('postcss-selector-parser').Selector} Selector */
+/** @typedef {import('postcss-selector-parser').Pseudo} Pseudo */
+/** @typedef {import('postcss-selector-parser').Container} Container */
+
+/**
+ * @param {Node} node
+ * @returns {boolean}
+ */
+const isPseudoClass = (node) =>
+	buggyIsPseudoClass(node) &&
+	node.value.toLowerCase() !== ':first-line' &&
+	node.value.toLowerCase() !== ':first-letter';
+
+/**
+ * @param {Node} node
+ * @returns {boolean}
+ */
+const isSimpleSelector = (node) =>
+	isPseudoClass(node) ||
+	isAttribute(node) ||
+	isClassName(node) ||
+	isUniversal(node) ||
+	isIdentifier(node) ||
+	isTag(node);
+
+/**
+ * @param {Node} node
+ * @returns {boolean}
```

Comment:
```suggestion
 * @returns {node is Pseudo}
```

Reply:
This function is not checking whether this is a pseudo or not.
Check its four references.

Is it unclear? Should it be renamed to `isNotPseudoClass` or add a description in the jsdoc comment?

Reply:
`postcss-selector-parser`'s original `isPseudoClass()` type-guard function checks that a specified node is a `Pseudo` type or not. The below is the definition:

```ts
function isPseudoClass(node: any): node is Pseudo;
```

So, I think we can make our `isNot()` (and also `isPseudoClass()`) a type-guard function.
The naming of `isNot()` is not my concern, but feel free to rename it.